mllib/src/main/scala/org/apache/spark/mllib/clustering/LocalKMeans.scala

```diff
@@ -46,17 +46,20 @@ private[mllib] object LocalKMeans extends Logging {
 
     // Initialize centers by sampling using the k-means++ procedure.
     centers(0) = pickWeighted(rand, points, weights).toDense
+    val costArray = points.map { point =>
+      KMeans.fastSquaredDistance(point, centers(0))
+    }
+
     for (i <- 1 until k) {
-      // Pick the next center with a probability proportional to cost under current centers
-      val curCenters = centers.view.take(i)
-      val sum = points.view.zip(weights).map { case (p, w) =>
-        w * KMeans.pointCost(curCenters, p)
+      val sum = costArray.view.zip(weights).map{
```

Comment:
Can be one line, I think, and needs space before brace. Maybe as simple to just `.map(p => p._1 * p._2)` but a case is OK too. I don't think a `.view` on a local array helps.